pkg/components/ldap_auth.go

```diff
@@ -0,0 +1,69 @@
+package components
+
+import (
+	"context"
+	"net/http"
+)
+
+type authValidationTransport struct {
+	Wrapped              http.RoundTripper
+	AllowedGroups        []string
+	LdapGroupsHeaderName string
+}
+
+func Contains(s []string, target string) bool {
+	for _, c := range s {
+		if target == c {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *authValidationTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	incomingLdapGroups := req.Header[r.LdapGroupsHeaderName]
+	allowedList := r.AllowedGroups
```

Comment:
suggestion: You don't need to do this. So long as all the use cases are read-only then you can use `r.AllowedGroups` directly without creating an alias here. This assignment makes a copy of the slice header but both variables are reading from the same underlying array.